lib/membrane_h264_ffmpeg/encoder.ex

```diff
@@ -68,14 +68,20 @@ defmodule Membrane.H264.FFmpeg.Encoder do
                 to decode video encoded with this element.
                 """,
                 type: :atom,
-                spec: H264.profile_t(),
-                default: :high
+                spec: H264.profile_t() | :auto,
+                default: :auto
```

Comment:
I think we can use `nil` instead of `:auto` here as well, but with a docs change. Investigating FFmpeg's options code (http://ffmpeg.org/doxygen/trunk/libx264_8c_source.html#l01089) and encoder help:
```
   --profile <string>      Force the limits of an H.264 profile
                                  Overrides all settings.
                                  - baseline:
                                    --no-8x8dct --bframes 0 --no-cabac
                                    --cqm flat --weightp 0
                                    No interlaced.
                                    No lossless.
                                  - main:
                                    --no-8x8dct --cqm flat
                                    No lossless.
                                  - high:
                                    No lossless.
                                  - high10:
                                    No lossless.
                                    Support for bit depth 8-10.
                                  - high422:
                                    No lossless.
                                    Support for bit depth 8-10.
                                    Support for 4:2:0/4:2:2 chroma subsampling.
                                  - high444:
                                    Support for bit depth 8-10.
                                    Support for 4:2:0/4:2:2/4:4:4 chroma subsampling.

```
This is actually a restriction and the lower profile can be achieved by setting other options, so in that case, the `nil` makes sense and maps to behaviour in native code (`nil` meaning 'not set')